### ISSUE TITLE
Duplicate ProcessedByFody Attribute when using dotnet publish

### ISSUE THREAD

**Issue by Inspyro** [Fody/Fody#895]
**Disclaimer: I'm not a backer but I think the severity of this issue warrents a bug report. This issue affects all users of Fody that use dotnet publish - i.e. for deploying a Web-Application or for creating a NuGet package**

When using **dotnet publish twice** the *_ProcessedByFody class is added multiple times to the resulting Assembly.
This in turn leads to a **System.BadImageFormatException**: Duplicate type with name "*_ProcessedByFody" thus not allowing the dll to be loaded as it is invalid. 

This is only reproducable with 2 projects, a exe that references a libarary, when publishing the exe that references the library, the libary has a duplicate *_ProcessedByFody attribute

**Repo steps**
* Download the [Fody_DuplicateProcessedByFodyAttributeRepro.zip](https://github.com/Fody/Fody/files/4964522/Fody_DuplicateProcessedByFodyAttributeRepro.zip) (or alternatively create 2 dotnet core csproj files - one exe referencing a library, add Fody + a fody weaver such as NullGuard.Fody)
* Navigate to Folder HelloFody/HelloFody
* Execute dotnet publish
 * Open the _HelloFody\HelloFody\bin\Debug\netcoreapp3.1\publish\SomeLibrary.dll_ and verify that the *_ProcessedByFody attribute exists only once (which is correct)
* Execute dotnet publish again
  * Note that now the *_ProcessedByFody attribute is in the dll twice, each time the dotnet publish command is executed another *\__ProcessedByFody_ attribute is added

*IlSpy Result after executing dotnet_publish twice*
![image](https://user-images.githubusercontent.com/7766101/88263935-2afef200-cccb-11ea-9999-dc733859145a.png)

**Comment by Inspyro** [comment]
@SimonCropp it seems as though this commit: https://github.com/Fody/Fody/pull/868/commits/45f696bbc8b65dede51769aeac37fced3360b915
caused the problem because it forgot to adjust the check: https://github.com/Fody/Fody/blob/8166778ec154416a97d4adf8c96a6396238ee3e7/FodyIsolated/InnerWeaver.cs#L99

**Comment by ltrzesniewski** [comment]
Maybe we should hook Fody to [`TargetsTriggeredByCompilation`](https://github.com/dotnet/roslyn/blob/12f1b8f6c17c0bcd271525b5f4db0f621944c49d/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L155), I guess that would have avoided this issue.

**Comment by SimonCropp** [comment]
@ltrzesniewski looks like it is worth a try

**Comment by ltrzesniewski** [comment]
Ok, I can do that if you'd like, but not before August.
In the meantime, I suggest we fix the line that @Inspyro pointed out (but I won't be able to submit a PR right now).

**Comment by drauch** [comment]
I'm affected too, I took the liberty of creating a pull request for you: https://github.com/Fody/Fody/pull/897

**Comment by ltrzesniewski** [comment]
I thought it would be possible to publish a release from my phone... it wasn't a very good idea. 😉 
Anyway, it should be fixed in v6.2.1.

**Comment by matkoch** [comment]
I'm getting a bunch of warnings still:

```
MSBUILD : warning : Fody: The assembly has already been processed by Fody. Weaving aborted. Path: XXX
```

Any way to disable those?

**Comment by ltrzesniewski** [comment]
@matkoch this most definitely shouldn't happen. Could you provide a repro (preferably in a new issue) please?

**Comment by matkoch** [comment]
@ltrzesniewski actually found another issue for this: https://github.com/Fody/Fody/issues/1023

Proposed fix doesn't resolve it though :/

**Comment by ltrzesniewski** [comment]
I wouldn't try to ignore this, as it suggests something went horribly wrong. I could change the way Fody is hooked up in the SDK as I suggested in a comment above, but I'd still like to know what went wrong. A repro would be appreciated 😁

**Comment by tom-englert** [comment]
#1023 is MAUI specific, but I guess this is something similar here, so a repo would be great.

**Comment by matkoch** [comment]
It's not minimal though :(

1. Checkout https://github.com/nuke-build/nuke/tree/feature/adaptive-logging
2. Execute `build.sh` or `build.ps1`
3. There should be an `msbuild.binlog` in the root

Note that calling `build.sh/ps1` will first compile all projects as part of compiling `_build.csproj`. After that, the `_build.csproj` will initiate another compilation. That's just what you end up when you dogfooding :)

**Comment by herebebeasties** [comment]
I think this is fixed in [6.7.0](https://www.nuget.org/packages/Fody/6.7.0) by the way.
